ui/src/app/pages/homepage/quick-tour-and-videos.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+
+import {Clickable} from 'app/components/buttons';
+import {FlexColumn, FlexRow} from 'app/components/flex';
+import {ClrIcon} from 'app/components/icons';
+import {Modal} from 'app/components/modals';
+import {Scroll} from 'app/icons/scroll';
+import colors, {addOpacity} from 'app/styles/colors';
+import {reactStyles} from 'app/utils';
+import {AnalyticsTracker} from 'app/utils/analytics';
+import {useEffect, useRef, useState} from 'react';
+import {QuickTourReact} from './quick-tour-modal';
+
+export const styles = reactStyles({
+  quickTourCardsRow: {
+    justifyContent: 'flex-start',
+    maxHeight: '26rem',
+    marginTop: '0.5rem',
+    marginBottom: '1rem',
+    marginLeft: '-1rem',
+    paddingLeft: '1rem',
+    position: 'relative'
+  },
+  quickTourAndVideosLabel: {
+    fontSize: 22,
+    lineHeight: '34px',
+    color: colors.primary,
+    paddingRight: '2.3rem',
+    fontWeight: 600,
+    marginTop: '1rem',
+    width: '33%'
+  },
+});
+
+interface Props {
+  showQuickTourInitially: boolean;
+}
+
+export const QuickTourAndVideos = (props: Props) => {
+  // my first attempt did not work as expected: useState(props.showQuickTourInitially)
+  const [showQuickTour, setShowQuickTour] = useState(false);
+  useEffect(() => {
+    setShowQuickTour(props.showQuickTourInitially);
+  }, [props]);
+
+  const [quickTourResourceOffset, setQuickTourResourceOffset] = useState(0);
+
+  const [videoId, setVideoId] = useState('');
+  const [showVideo, setShowVideo] = useState(false);
+
+  const openVideo = (id: string) => {
+    AnalyticsTracker.Registration.TutorialVideo();
+    setVideoId(id);
+    setShowVideo(true);
+  };
+
+  // The videoId parameters below are the YouTube ids that get inserted into the src url of the iframe
+  const quickTourResources = [
+    {
+      src: '/assets/images/QT-thumbnail.svg',
+      onClick: () => setShowQuickTour(true)
+    }, {
+      src: '/assets/images/intro-workbench.png',
+      onClick: () => openVideo('NTLJtwLcavo')
+    }, {
+      src: '/assets/images/cohort-builder.png',
+      onClick: () => openVideo('G6_GG2CJ9mA')
+    }, {
+      src: '/assets/images/dataset-builder.png',
+      onClick: () => openVideo('cUuDKUxjQoo')
+    }, {
+      src: '/assets/images/notebook-code-snippets.png',
+      onClick: () => openVideo('NvMWBlVyyUU')
+    }, {
+      src: '/assets/images/user-support.png',
+      onClick: () => openVideo('Ni4PEJVbmSk')
+    }
+  ];
+
+  // This calculates the limit for quickTourResources items that can be seen without scrolling. Takes the width of the parent element
+  // and divides by the width of an individual resource item (276px). The default limit is 4 since the min width of the parent element
+  // should be ~1128px
+  const quickTourResourcesDiv = useRef<HTMLDivElement>(null);
```

Comment:
I'm not solid on refs so it would be good to pay extra attention here

Reply:
Yes: it will display 4, 5, or 6 depending on screen width on initial load.  Matching existing behavior, it won't go below 4.